app/src/main/java/com/lastaoutdoor/lasta/ui/screen/discover/components/RangeSearchComposable.kt

```diff
@@ -1,198 +1,202 @@
-                      modifier = Modifier.width(300.dp).testTag("mapRangeSearchSlider"))
-                  Text(
-                      // put range in km
-                      text = "${(range / 1000).toInt()}km",
-                      style = MaterialTheme.typography.bodyMedium,
-                      modifier = Modifier.padding(8.dp))
-                }
-
-                // Search bar to search by locality
-                Row {
-                  Text(
-                      text = LocalContext.current.getString(R.string.locality),
-                      style =
-                          TextStyle(
-                              fontSize = 16.sp, lineHeight = 24.sp, fontWeight = FontWeight(500)))
-                  Spacer(modifier = Modifier.height(8.dp))
-                  // Dropdown to select the city
-                  LocalitySelectionDropdown(localities, selectedLocality, setSelectedLocality)
-                }
-                Spacer(modifier = Modifier.height(12.dp))
-
-                // option to use current location with blue text and location icon
-                Row(
-                    modifier =
-                        Modifier.clickable { /*TODO SET SELECTEDLOCALITY TO CURRENT POSITION */}) {
-                      Icon(
-                          imageVector = Icons.Default.LocationOn,
-                          contentDescription = "Settings",
-                          tint = MaterialTheme.colorScheme.primary)
-                      Text(
-                          text = LocalContext.current.getString(R.string.use_curr_loc),
-                          style =
-                              TextStyle(
-                                  fontSize = 16.sp,
-                                  lineHeight = 24.sp,
-                                  fontWeight = FontWeight(500),
-                                  color = PrimaryBlue))
-                    }
-
-                Spacer(modifier = Modifier.height(12.dp))
-
-                // Button to apply the range
-                ElevatedButton(
-                    onClick = {
-                      discoverScreenCallBacks.updateRange(range)
-                      discoverScreenCallBacks.updateInitialPosition(selectedLocality.second)
-                      onDismissRequest()
-                    },
-                    modifier =
-                        Modifier.width(305.dp).height(48.dp).testTag("mapRangeSearchApplyButton"),
-                    colors = ButtonDefaults.buttonColors(containerColor = PrimaryBlue)) {
-                      Text(
-                          LocalContext.current.getString(R.string.apply),
-                          style =
-                              TextStyle(
-                                  fontSize = 22.sp,
-                                  lineHeight = 28.sp,
-                                  fontWeight = FontWeight(400),
-                              ))
-                    }
-              }
-        }
+      Canvas(
```

Comment:
I think this is the first time we use this composable. Nice implementation, I like it!